app/screens/home/bottomTabs/Articles.tsx

```diff
@@ -396,100 +485,127 @@ const Articles = ({ route, navigation }: any): any => {
 
     }, [route.params?.categoryArray, activeChild?.uuid, languageCode, queryText])
   );
-  useFocusEffect(
-    React.useCallback(() => {
-      console.log('UseFouusEffect Articles two');
-      const showSubscription = Keyboard.addListener("keyboardDidShow", () => {
-        setKeyboardStatus(true);
-        //  setHistoryVisible(true);
-      });
-      const hideSubscription = Keyboard.addListener("keyboardDidHide", () => {
-        setKeyboardStatus(false);
-      });
-      return (): any => {
-        {
-          navigation.setParams({ categoryArray: [] })
-          showSubscription.remove();
-          hideSubscription.remove();
-          // route.params?.currentSelectedChildId = 0;
-        }
-      }
-    }, [])
-  );
 
+  const [searchIndex, setSearchIndex] = useState<any>(null);
+  const suffixes = (term: any, minLength: any): any => {
+    if (term == null) { return []; }
+    const tokens = [];
+    for (let i = 0; i <= term.length - minLength; i++) {
+      tokens.push(term.slice(i));
+    }
+    return tokens;
+  }
+  useEffect(() => {
+    async function initializeSearchIndex() {
+
+      const videoarticleDataAllCategory = VideoArticlesDataall.filter((x: any) => x.mandatory == videoArticleMandatory && x.child_age.includes(activeChild.taxonomyData.id) && (x.child_gender == activeChild?.gender || x.child_gender == bothChildGender));

```

Comment:
Let's check activeChild and its properties for null or undefined to prevent errors.